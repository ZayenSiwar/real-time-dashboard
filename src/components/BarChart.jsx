import { Bar } from "react-chartjs-2";
import {
  Chart as ChartJS,
  CategoryScale,
  LinearScale,
  BarElement,
  Title,
  Tooltip,
  Legend
} from "chart.js";

ChartJS.register(
  CategoryScale,
  LinearScale,
  BarElement,
  Title,
  Tooltip,
  Legend
);

export default function BarChart() {
  const data = {
    labels: ["Jan", "FÃ©v", "Mar", "Avr", "Mai", "Juin", "Juil"],
    datasets: [
      {
        label: "Ventes",
        data: [30, 50, 40, 60, 70, 55, 80],
        backgroundColor: "rgba(255, 206, 86, 0.7)",
      },
    ],
  };

  const options = {
    responsive: true,
    plugins: {
      legend: { position: "top" },
      title: { display: true, text: "Ventes mensuelles" },
    },
  };

  return <Bar data={data} options={options} />;
}
