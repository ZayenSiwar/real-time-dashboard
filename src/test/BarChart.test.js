import React from 'react';
import { render, screen } from '@testing-library/react';
import BarChart from '../components/BarChart';

// Mock Chart.js and react-chartjs-2
jest.mock('react-chartjs-2', () => ({
  Bar: ({ data, options }) => (
    <div data-testid="bar-chart">
      <div data-testid="chart-data">{JSON.stringify(data)}</div>
      <div data-testid="chart-options">{JSON.stringify(options)}</div>
    </div>
  ),
}));

describe('BarChart Component', () => {
  it('renders without crashing', () => {
    render(<BarChart />);
    expect(screen.getByTestId('bar-chart')).toBeInTheDocument();
  });

  it('displays correct chart data structure', () => {
    render(<BarChart />);
    const chartData = JSON.parse(screen.getByTestId('chart-data').textContent);
    
    expect(chartData).toHaveProperty('labels');
    expect(chartData).toHaveProperty('datasets');
    expect(Array.isArray(chartData.labels)).toBe(true);
    expect(Array.isArray(chartData.datasets)).toBe(true);
  });

  it('has correct month labels in French', () => {
    render(<BarChart />);
    const chartData = JSON.parse(screen.getByTestId('chart-data').textContent);
    
    expect(chartData.labels).toEqual([
      "Jan", "FÃ©v", "Mar", "Avr", "Mai", "Juin", "Juil"
    ]);
  });

  it('has correct dataset configuration', () => {
    render(<BarChart />);
    const chartData = JSON.parse(screen.getByTestId('chart-data').textContent);
    
    expect(chartData.datasets[0]).toHaveProperty('label', 'Ventes');
    expect(chartData.datasets[0]).toHaveProperty('data');
    expect(chartData.datasets[0]).toHaveProperty('backgroundColor', 'rgba(255, 206, 86, 0.7)');
  });

  it('has correct chart options', () => {
    render(<BarChart />);
    const chartOptions = JSON.parse(screen.getByTestId('chart-options').textContent);
    
    expect(chartOptions).toHaveProperty('responsive', true);
    expect(chartOptions).toHaveProperty('plugins');
    expect(chartOptions.plugins).toHaveProperty('legend');
    expect(chartOptions.plugins).toHaveProperty('title');
    expect(chartOptions.plugins.title).toHaveProperty('text', 'Ventes mensuelles');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<BarChart />);
    expect(asFragment()).toMatchSnapshot();
  });
});
