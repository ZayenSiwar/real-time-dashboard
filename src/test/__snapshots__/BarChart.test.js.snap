// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BarChart Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="bar-chart"
  >
    <div
      data-testid="chart-data"
    >
      {"labels":["Jan","FÃ©v","Mar","Avr","Mai","Juin","Juil"],"datasets":[{"label":"Ventes","data":[30,50,40,60,70,55,80],"backgroundColor":"rgba(255, 206, 86, 0.7)"}]}
    </div>
    <div
      data-testid="chart-options"
    >
      {"responsive":true,"plugins":{"legend":{"position":"top"},"title":{"display":true,"text":"Ventes mensuelles"}}}
    </div>
  </div>
</DocumentFragment>
`;
